src/org/parosproxy/paros/core/scanner/HostProcess.java

```diff
@@ -555,22 +556,66 @@ public boolean handleAntiCsrfTokens() {
     }
 
     /**
-     * Skip the current executing plugin
-     * @param plugin the plugin instance that need to be skipped
+     * Skips the given plugin.
+     * <p>
+     * <strong>Note:</strong> Whenever possible it should be used {@link #pluginSkipped(Plugin, String)} instead.
+     * 
+     * @param plugin the plugin that will be skipped, must not be {@code null}
+     * @since 2.4.0
      */
     public void pluginSkipped(Plugin plugin) {
-        if (pluginFactory.isRunning(plugin)) {
-            listPluginIdSkipped.add(plugin.getId());
+        pluginSkipped(plugin, null);
+    }
+
+    /**
+     * Skips the given {@code plugin} with the given {@code reason}.
+     * <p>
+     * Ideally the {@code reason} should be internationalised as it is shown in the GUI.
+     *
+     * @param plugin the plugin that will be skipped, must not be {@code null}
+     * @param reason the reason why the plugin was skipped, might be {@code null}
+     * @since TODO add version
+     */
+    public void pluginSkipped(Plugin plugin, String reason) {
+        PluginStats pluginStats = mapPluginStats.get(plugin.getId());
+        if (pluginStats == null) {
+            return;
         }
+
+        pluginStats.skipped();
+        pluginStats.setSkippedReason(reason);
     }
 
     /**
-     * Check if a specific plugin has been explicitly skipped by the user 
-     * @param plugin the plugin instance currently running
-     * @return true if the user has skipped this instance
+     * Tells whether or not the given {@code plugin} was skipped (either programmatically or by the user).
+     * 
+     * @param plugin the plugin that will be checked
+     * @return {@code true} if plugin was skipped, {@code false} otherwise
+     * @since 2.4.0
+     * @see #getSkippedReason(Plugin)
      */
     public boolean isSkipped(Plugin plugin) {
-        return (!listPluginIdSkipped.isEmpty() && listPluginIdSkipped.contains(plugin.getId()));
+        PluginStats pluginStats = mapPluginStats.get(plugin.getId());
+        if (pluginStats == null) {
+            return false;
+        }
+        return pluginStats.isSkipped();
+    }
+
+    /**
+     * Gets the reason why the given plugin was skipped.
+     * 
+     * @param plugin the plugin that will be checked
+     * @return the reason why the given plugin was skipped, might be {@code null} if not skipped or there's no reason

```

Comment:
Oh ok, leave it as is. I was only consindering the return comment in the javadoc, if null becomes generic "Skipped" elsewhere then the condition is handled. From a support perspective that gives the differentiation I was concerned about.